src/spicy/manager.cc

```diff
@@ -47,6 +47,19 @@ static std::pair<std::string, std::string> parseID(const std::string& s) {
 
 Manager::~Manager() {}
 
+void Manager::registerSpicyModuleBegin(const std::string& name, const std::string& description, hilti::rt::Time mtime) {
+    _module_info =
+        std::make_unique<zeekygen::detail::SpicyModuleInfo>(name, description, static_cast<time_t>(mtime.seconds()));
```

Comment:
This doesn't handle (accidental?) nesting of calls to `register_spicy_module_begin`. At a minimum we should `assert(!_module_info)` here.

That being said, I wonder whether we need the complications of `begin`/`end` functions at all. Couldn't we instead if needed pass a module (e.g., via ID) to the various register functions so can trigger accumulation of state and then only have a single commit function by ID (currently in the end function)?

Reply:
I'll add the `assert`, nesting shouldn't be able to occur.

I thought about adding new parameters to the register functions, but figured bracketing them with a pair of functions enabling tracking of module info, would actually be easier / less complex.